src/lib/components/AddOrUpdateNote.svelte

```diff
@@ -25,62 +27,56 @@
 
 	let success = $state(false);
 	let errorMessage = $state<string>("");
+	let editor = $state<Editor>();
 
-	async function handleSubmit() {
+	async function handleSubmit(e?: MouseEvent) {
+		e?.preventDefault();
 		errorMessage = "";
 		// if the note is already in the db, do not change the created date
-		var createdDate: Date;
-		if (note?.id && note.createdDate) {
-			createdDate = note.createdDate;
-		} else {
-			createdDate = new Date();
-		}
-		note.createdDate = createdDate;
+		note.createdDate =
+			note?.id && note.createdDate ? note.createdDate : new Date();
 		if (!note.title) {
 			errorMessage = "Please add a title to your note";
 			return;
 		}
-		const newNote: Note = {
-			...note,
-		};
-
-		success = await addOrUpdateNote(newNote);
-
+		note.content = editor?.getHTML() ?? note.content;
+		success = await addOrUpdateNote({ ...note });
 		if (success) {
 			open = false;
-			note = { ...DEFAULT_NOTE };
 		}
-		onupdate && onupdate(newNote);
+		onupdate && onupdate({ ...note });
 	}
 </script>
 
-{#if open}
-	<Modal size="sm" title={note.id ? "Edit Note" : "New Note"} bind:open>
-		<TextInput label="Title" bind:value={note.title} />
-		<TextInput type="textarea" label="Content" bind:value={note.content} />
+<Modal size="sm" title={note.id ? "Edit Note" : "New Note"} bind:open>
+	<TextInput label="Title" bind:value={note.title} />
+	<TipTap
+		content={note.content ?? ""}
+		onupdate={() => handleSubmit}
+		bind:editor
+	/>
 
-		{#if errorMessage}
-			<div class="text-red-500 m-2">{errorMessage}</div>
-		{/if}
+	{#if errorMessage}
+		<div class="text-red-500 m-2">{errorMessage}</div>
+	{/if}
 
-		<div class="flex justify-between">
-			<div class="flex">
-				<input
-					oninput={(e) => {
-						note.dueDate = new Date();
-						const tempDate = e.currentTarget.value.split("-");
-						note.dueDate.setFullYear(parseInt(tempDate[0]));
-						note.dueDate.setMonth(parseInt(tempDate[1]) - 1);
-						note.dueDate.setDate(parseInt(tempDate[2]));
-					}}
-					value={note.dueDate?.toISOString().split("T")[0]}
-					type="date"
-					id="date"
-					class="rounded-lg border border-gray-300 bg-gray-800 text-gray-100 focus:border-transparent focus:ring-2 focus:ring-blue-500 focus:outline-none"
-				/>
-			</div>
-
-			<Button onclick={handleSubmit}>Submit</Button>
+	<div class="flex justify-between">
```

Comment:
<div class="flex justify-between mt-2">

adds spacing between the bottom of the tiptap editor and the date and button elements. will look much better.